src/active_sound.h

```diff
@@ -1,27 +1,34 @@
 #ifndef ACTIVE_SOUND_H
 #define ACTIVE_SOUND_H
 
-#include "enums.h"
 #include "filters/filters.h"
+#include "enums.h"
 #include "soloud.h"
 
 #include <iostream>
 #include <vector>
 #include <memory>
 
 class Filters;
+#define MAX_DOUBLE 1.7976931348623157e+308
+
+struct ActiveHandle
+{
+    SoLoud::handle handle;
+    SoLoud::time bufferingTime;
+};
 
 /// The default number of concurrent voices - maximum number of "streams" - is 16,
 /// but this can be adjusted at runtime
-typedef struct ActiveSound
+struct ActiveSound
 {
     std::unique_ptr<SoLoud::AudioSource> sound;
     SoundType soundType;
-    std::vector<SoLoud::handle> handle;
+    std::vector<ActiveHandle> handle;
```

Comment:
I'm starting to miss documentation here. Is this the list of sound handles that are currently pointing to (i.e. playing) this `ActiveSound`?

Reply:
You are correct. That `handle` is still a list of handles currently active in the `AudioSource`, but I had to add a parameter for each handle to know if/when the handle has reached the end of the data to catch the buffering event. This is managed only in the new `audiobuffer.cpp` and it doesn't affect anything else.
Added a comment here.